docs/api.rst

```diff
@@ -1,11 +1,6 @@
 API documentation
 -----------------
 
-.. autoclass:: pytket.extensions.cutensornet.CuTensorNetHandle
-
-    .. automethod:: destroy
-
-
```

Comment:
No longer used by the `general_state` module, so it makes more sense that it only appears in the docs of `structured_state`.